### PR TITLE
Fix mailchimp signup bug

### LINKED ISSUE #965: Home: mailchimp signup needs to use https
It works on the current marketing site, because that's only on http

```
jquery-2.2.0.min.js:4 Mixed Content: The page at 'https://micromasters-ci.herokuapp.com/' was loaded over HTTPS, but requested an insecure script 'http://facebook.us6.list-manage.com/subscribe/post-json?u=ad81d725159c1f322…ersity+or+educational+institution&MMERGE4=&MMERGE3=MIT+ODL&_=1472821806637'. This request has been blocked; the content must be served over HTTPS.send @ jquery-2.2.0.min.js:4ajax @ jquery-2.2.0.min.js:4(anonymous function) @ jquery.ajaxchimp.js:120dispatch @ jquery-2.2.0.min.js:3r.handle @ jquery-2.2.0.min.js:3
jquery.ajaxchimp.js:126 mailchimp ajax submit error: error
```

The code in question is here: https://github.com/mitodl/micromasters/blob/536a36729e633979bccc1f3d30c23298213de4c4/static/js/public.js#L71

### PR DESCRIPTION
**Pull request by annagav**
#### What are the relevant tickets?
Fix #965 
#### What's this PR do?
Fix the Mixed Content error.

To test this, go to the home page through this link
https://micromasters-ci-pr-1055.herokuapp.com/.
Try to sign up for the mailing list. On submit the error used to occur (you can see it at https://micromasters.herokuapp.com/)

### CODE REVIEW

File: cms/templates/cms/home_page.html [hunk 1]
```diff
@@ -160,7 +160,7 @@
             <div class="program-thumbnail-description">
               <p>MIT plans to bring more MicroMasters programs to motivated learners like you.
                 Sign up for our mailing list to hear about new MicroMasters programs.</p>
-              <button type="button" class="btn white-button">SIGN UP FOR MAILING LIST</button>
+              <a href="#" data-toggle="modal" class="btn white-button" data-target="#subscribeAlert">SIGN UP FOR MAILING LIST</a>
```
Comment (pdpinch) [inline cms/templates/cms/home_page.html]: Why did you make this change? Just curious.

Comment (annagav) [inline cms/templates/cms/home_page.html]: I realized this is how it was already implemented in the past.

**Comment by annagav** [comment]
@pdpinch, Is it ready for merge?

**Comment by pdpinch** [comment]
👍 

I used the new GitHub flow for approving PRs. Maybe it's less obvious, but you should see something like this above the merge button:

screen shot 2016-09-16 at 11 57 44 am

**Pull request merged**